conn.go

```diff
@@ -16,11 +16,47 @@
 
 package edgedb
 
+import (
+	"context"
+
+	"github.com/edgedb/edgedb-go/internal/cache"
+)
+
 // Conn is a conn created outside of a pool.
 type Conn struct {
 	baseConn
 }
 
+// ConnectOne establishes a connection to an EdgeDB server.
+func ConnectOne(ctx context.Context, opts Options) (*Conn, error) { // nolint:gocritic,lll
```

Comment:
I'm just curious what kind of linting rule does `nolint:gocritic,lll` disable?

Reply:
This repo is using a meta linter [golangci-lint](https://github.com/golangci/golangci-lint). The names are linters, [gocritic (general linter)](https://github.com/go-critic/go-critic) and [lll (line length linter)](https://github.com/walle/lll).

Reply:
Specifically, gocritic complains about passing `Options` by value (copies) rather than passing a pointer because `Options` is a large struct. However, I think that we want a copy here so that the user can't change the options later.